apps/opencs/view/render/brushdraw.cpp

```diff
@@ -0,0 +1,308 @@
+            getIntersectionHeight(osg::Vec3d(
+                point.x() - radius,
+                point.y() - radius + step2,
+                point.z())) + brushOutlineHeight);
+        osg::Vec3d downHorizontalLinePoint1(
+            point.x() + radius - step,
+            point.y() + radius,
+            getIntersectionHeight(osg::Vec3d(
+                point.x() + radius - step,
+                point.y() + radius,
+                point.z())) + brushOutlineHeight);
+        osg::Vec3d downHorizontalLinePoint2(
+            point.x() + radius - step2,
+            point.y() + radius,
+            getIntersectionHeight(osg::Vec3d(
+                point.x() + radius - step2,
+                point.y() + radius,
+                point.z())) + brushOutlineHeight);
+        osg::Vec3d downVerticalLinePoint1(
+            point.x() + radius,
+            point.y() + radius - step,
+            getIntersectionHeight(osg::Vec3d(
+                point.x() + radius,
+                point.y() + radius - step,
+                point.z())) + brushOutlineHeight);
+        osg::Vec3d downVerticalLinePoint2(
+            point.x() + radius,
+            point.y() + radius - step2,
+            getIntersectionHeight(osg::Vec3d(
+                point.x() + radius,
+                point.y() + radius - step2,
+                point.z())) + brushOutlineHeight);
+        vertices->push_back(upHorizontalLinePoint1);
+        colors->push_back(lineColor);
+        vertices->push_back(upHorizontalLinePoint2);
+        colors->push_back(lineColor);
+        vertices->push_back(upVerticalLinePoint1);
+        colors->push_back(lineColor);
+        vertices->push_back(upVerticalLinePoint2);
+        colors->push_back(lineColor);
+        vertices->push_back(downHorizontalLinePoint1);
+        colors->push_back(lineColor);
+        vertices->push_back(downHorizontalLinePoint2);
+        colors->push_back(lineColor);
+        vertices->push_back(downVerticalLinePoint1);
+        colors->push_back(lineColor);
+        vertices->push_back(downVerticalLinePoint2);
+        colors->push_back(lineColor);
+    }
+
+    geom->setVertexArray(vertices);
+    geom->setColorArray(colors, osg::Array::BIND_PER_VERTEX);
+    geom->addPrimitiveSet(new osg::DrawArrays(osg::PrimitiveSet::LINES, 0, resolution * 8));
+    mGeometry = geom;
+}
+
+void CSVRender::BrushDraw::buildCircleGeometry(const float& radius, const osg::Vec3d& point)
+{
+    osg::ref_ptr<osg::Geometry> geom = new osg::Geometry();
+    osg::ref_ptr<osg::Vec3Array> vertices (new osg::Vec3Array());
+    osg::ref_ptr<osg::Vec4Array> colors (new osg::Vec4Array());
+    const int amountOfPoints = (osg::PI * 2.0f) * radius / 20;
+    const float step ((osg::PI * 2.0f) / static_cast<float>(amountOfPoints));
+    const float brushOutlineHeight (1.0f);
+    osg::Vec4f lineColor(1.0f, 1.0f, 1.0f, 0.6f);
+
+    for (int i = 0; i < amountOfPoints + 2; i++)
+    {
+        float angle (static_cast<float>(i) * step);
```

Comment:
This static_cast should be unnecessary, since step is a float. int * float is a float.

Reply:
Right... I've had some trouble occasionally with calculations having both float and int, static_cast was done here to enforce that I get a float calculation and I can be absolutely sure this works. Removing that should be anyway a minor optimization and easier code to read, so I'll do that.